src/poc/server/poc-server/src/controllers/users.ts

```diff
@@ -12,7 +17,50 @@ export const createSession = async (req: Request, res: Response) => {
     const uuid: string = uuidv4();
     console.log(uuid);
     sessionIds.push(uuid);
-    res.status(200).json({uuid});
+    res.status(200).json({ uuid });
+}
+
+/**
+ * Route to verify a users location
+ * @name /api/users/verify-location
+ * @function
+ */
+export const verifyLocation = async (req: Request, res: Response) => {
+    const uuid: string = req.body.userId;
+    try {
+        if (!sessionIds.includes(uuid)) {
+            res.status(401).json({ message: "User does not exist" });
+            return;
+        }
+
+        console.log(
+            "User [%s] exists at [%s] longitude and [%s] latitude",
+            uuid,
+            req.body.longitude,
+            req.body.latitude
+        );
+        const userLocation = {
+            latitude: req.body.latitude,
+            longitude: req.body.longitude
+        };
+
+        const distance: number = haversine(userLocation, ITBLocation)
+
+        console.log("User [%s], is [%d] meters from the site", uuid, distance)
+
+        if (distance > 10) {
+            console.log(
+                "User [%s], is too far from the site, failed to authenticate",
+                uuid
+            );
+            res.status(401).json({ message: "User is not on a registered site" });
+            return;
+        }
+        res.status(200).json({ token: "" });
```

Comment:
did we want to return a token from this? @Spitgranger @akshitg3

Reply:
Ya I think that's how we're gonna tell if a user is authorized or not after they provide their location